runtime/unix.c

```diff
@@ -440,6 +440,16 @@ int caml_num_rows_fd(int fd)
 #endif
 }
 
+void caml_thread_getname(char* name)
+{
+  pthread_t self = pthread_self();
+#if defined(__OpenBSD__) || defined(__FreeBSD__)
+  pthread_get_name_np(self, name, 16);
+#else /* linux glibc/musl or NetBSD or apple */
+  pthread_getname_np(self, name, 16);
```

Comment:
again here I'm not sure why we are hardcoding 16; could we have a named constant instead?